plugin/src/test/resources/com/google/errorprone/xplat/checker/testdata/LazyInitBanNegativeCases.java

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.errorprone.xplat.checker.testdata;
+
+
+import autovalue.shaded.com.google$.errorprone.annotations.concurrent.$LazyInit;
+import com.google.errorprone.annotations.concurrent.LazyInit;
+
+public class LazyInitBanNegativeCases {
+
+  private final int x;
+  private volatile String y;
+  @LazyInit
+  private String z;
+
+  private String zz;
+
+  private String a = "hello";
+
+  // syncronized with volatile
+  public synchronized String lazyInitValid() {
+    if (y == null) {
+      y = new String();
+    }
+    return y;
+  }
+
+  // syncronized with @LazyInit
+  public synchronized String lazyInitValid1() {
+    if (z == null) {
+      z = new String();
+    }
+    return z;
+  }
+
+  //syncronized inside
+  public String lazyInitValid2() {
+    synchronized (this) {
+      if (y == null) {
+        y = new String();
+      }
+      return y;
+    }
+  }
+
+  //syncronized inside
+  public String lazyInitValid3() {
+    synchronized (this) {
+      if (z == null) {
+        z = new String();
+      }
+      return z;
+    }
+  }
+
+  // non-sync version
+  public String lazyInitValid4() {
+    String local = y;
```

Comment:
can any lines exist between the creation of the local and the test?

Reply:
I think this falls under a question whether we should tighten the bounds of the checker so much so that it becomes a template checker: "we only allows you to implement single-check lazy inits with these only allowed patterns". Even that there will still be some variations (do we allow ternary operators? and so on)…

Reply:
If random other lines exist, it will currently just not give an error.